src/components/BucketItems.js

```diff
@@ -80,12 +80,34 @@ class BucketItems extends Component {
                     })
 
                 }
+
+                handleSearchItem = (searchText) => {
+                    axios.get(BASE_URL + `/bucketlist/${this.props.match.params.id}/item?q=`+searchText,
+                        {headers: {
+                            "Authorization": localStorage.getItem('token'),
+                            "content-Type":'application/json'
+                        }
+                        }).then(response => {
+                            this.setState({
+                            items :response.data.item
+                        })      
+                })
+                }
             
             // Gets all the bucketlist items before the page renders
             componentWillMount(){
```

Comment:
Have any `React Lifecycle` hooks immediately after the constructor. It is best practice.